core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SparqlSession.java

```diff
@@ -172,6 +188,24 @@ public SparqlSession(HttpClient client, ExecutorService executor) {
 		// parser used for processing server response data should be lenient
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_DATATYPE_VALUES);
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_LANGUAGE_TAGS);
+
+		// configure the maximum url length for SPARQL query GET requests
+		int maximumUrlLength = DEFAULT_MAXIMUM_URL_LENGTH;
+		String propertyValue = System.getProperty(MAXIMUM_URL_LENGTH_PARAM);
+		if (propertyValue != null) {
+			try {
+				maximumUrlLength = Integer.parseInt(propertyValue);
+			}
+			catch (NumberFormatException e) {
+				throw new RDF4JException("integer value expected for property " + MAXIMUM_URL_LENGTH_PARAM,
+						e)
+				{
+
+					private static final long serialVersionUID = 1L;
```

Comment:
It needs to happen this way because RDF4JException is abstract: this code creates a concrete anonymous subclass (which of course needs its own serialization version number). Admittedly a bit cavalier to create an exception in this way but the problem is not in RDF4JException itself (unless you consider it being abstract a problem, of course :))